mbedtls/src/mbedtls_malloc.c

```diff
@@ -21,11 +21,15 @@
 #define mbedtls_free      free
 #endif
 
-extern void *forward_mbedtls_calloc( size_t n, size_t size ) {
+// Use several macros to get the preprocessor to actually replace RUST_MBEDTLS_METADATA_HASH
+#define append_macro_inner(a, b) a##_##b
+#define append_macro(a, b) append_macro_inner(a, b)
```

Comment:
Never mind, I got it:

> The first macro append_macro_inner concatenates two tokens together using the ## operator. However, since RUST_MBEDTLS_METADATA_HASH is an environment variable and not a token, it needs to be expanded by the preprocessor before it can be concatenated.